packages/wdio-shared-store-service/src/server.ts

```diff
@@ -46,6 +49,44 @@ export const startServer = () => new Promise<{ port: number, app: PolkaInstance
             store[key] = req.body.value as JsonCompatible | JsonPrimitive
             return res.end()
         })
+        .post('/setResourcePool', (req, res) => {
+            const key = req.body.key as string
+            const value = req.body.value as JsonCompatible | JsonPrimitive
+
+            if (!Array.isArray(value)) {
+                throw new Error('Resource pool must be an array of values')
+            }
+
+            resourcePoolStore[key] = value
+            return res.end()
+        })
+        .post('/takeValueFromPool', (req, res) => {
+            const key = req.body.key as string
+            const pool = resourcePoolStore[key]
+
+            if (!pool) {
+                throw new Error(`'${key}' resource pool is does not exist. Set it first`)
+            }
+
+            if (pool.length === 0) {
+                throw new Error(`'${key}' resource pool is empty. Return values to it first`)
+            }
+
+            const value = pool.shift()
```

Comment:
I've tried this:

```javascript
.get('/getValueFromPool', (req, res) => {
            const key = req.body.key as string
            let value = getValueFromPool(key);

            if (value) {
                res.end(JSON.stringify({ value }))
                return
            }

            setTimeout(function secondAttempt() {
                value = getValueFromPool(key);
                if (!value) {
                    throw new Error(`'${key}' resource pool is empty. Set values to it first using 'setResourcePool'`)
                }
                res.end(JSON.stringify({ value }))
            }, 1000)
        })
```

It may be best for the user to specify the desired timeout amount, however this doesn't allow for further retries or faster polling. Should we let that down to the user?

Reply:
Implemented timeout, please check now